lib/msal-browser/src/broker/nativeBroker/PlatformAuthDOMHandler.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    Logger,
+    createAuthError,
+    AuthErrorCodes,
+    IPerformanceClient,
+    StringDict,
+} from "@azure/msal-common/browser";
+import {
+    PlatformBrokerRequest,
+    PlatformDOMTokenRequest,
+} from "./PlatformBrokerRequest.js";
+import { createNewGuid } from "../../crypto/BrowserCrypto.js";
+import { NativeConstants } from "../../utils/BrowserConstants.js";
+import {
+    PlatformBrokerResponse,
+    PlatformDOMTokenResponse,
+} from "./PlatformBrokerResponse.js";
+import { createNativeAuthError } from "../../error/NativeAuthError.js";
+import { IPlatformAuthHandler } from "./IPlatformAuthHandler.js";
+
+export class PlatformAuthDOMHandler implements IPlatformAuthHandler {
+    protected logger: Logger;
+    protected performanceClient: IPerformanceClient;
+    protected correlationId: string;
+    protected brokerId: string;
+    protected extensionVersion: string;
+    platformAuthType: string;
+
+    constructor(
+        logger: Logger,
+        performanceClient: IPerformanceClient,
+        correlationId?: string
+    ) {
+        this.logger = logger;
+        this.performanceClient = performanceClient;
+        this.brokerId = NativeConstants.MICROSOFT_ENTRA_BROKERID;
+        this.correlationId = correlationId || createNewGuid();
```

Comment:
nit: Along with the above
```suggestion
        this.correlationId = correlationId;
```